htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxReswap.java

```diff
@@ -0,0 +1,279 @@
+                showTarget, that.showTarget) && Objects.equals(focusScroll, that.focusScroll);
+    }
+
+    /**
+     * Enable or disable auto-scrolling to focused inputs between requests.
+     *
+     * @param enabled {@code true} if auto-scrolling to focused inputs between requests is enabled, otherwise {@code false}
+     * @return self
+     */
+    public HtmxReswap focusScroll(boolean enabled) {
+        this.focusScroll = enabled;
+        return this;
+    }
+
+    public Boolean getFocusScroll() {
+        return focusScroll;
+    }
+
+    public Position getScroll() {
+        return scroll;
+    }
+
+    public String getScrollTarget() {
+        return scrollTarget;
+    }
+
+    public Duration getSettle() {
+        return settle;
+    }
+
+    public Position getShow() {
+        return show;
+    }
+
+    public String getShowTarget() {
+        return showTarget;
+    }
+
+    public Duration getSwap() {
+        return swap;
+    }
+
+    public HxSwapType getType() {
+        return type;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(type, swap, settle, scroll, scrollTarget, show, showTarget, transition, focusScroll);
+    }
+
+    public boolean isTransition() {
+        return transition;
+    }
+
+    /**
+     * Changes the scrolling behavior of the target element.
+     *
+     * @param position top or bottom
+     * @return self
+     */
+    public HtmxReswap scroll(Position position) {
+        this.scroll = position;
+        return this;
+    }
+
+    /**
+     * Used to target a different element for scrolling.
+     *
+     * @param cssSelector a CSS selector
+     * @return self
+     */
+    public HtmxReswap scrollTarget(String cssSelector) {
+        this.scrollTarget = cssSelector;
+        return this;
+    }
+
+    /**
+     * Set the time that should elapse between the swap and the settle logic.
+     *
+     * @param duration the time to wait
+     * @return self
+     */
+    public HtmxReswap settle(Duration duration) {
+        this.settle = duration;
+        return this;
+    }
+
+    /**
+     * Changes the scrolling behavior of the target element.
+     *
+     * @param position top or bottom
+     * @return self
+     */
+    public HtmxReswap show(Position position) {
+        this.show = position;
+        return this;
+    }
+
+    /**
+     * Used to target a different element for showing.
+     *
+     * @param cssSelector a CSS selector
+     * @return self
+     */
+    public HtmxReswap showTarget(String cssSelector) {
+        this.showTarget = cssSelector;
+        return this;
+    }
+
+    /**
+     * Set the time that should elapse after receiving a response to swap the content.
+     *
+     * @param duration the time to wait
+     * @return self
+     */
+    public HtmxReswap swap(Duration duration) {
+        this.swap = duration;
+        return this;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Might want to use a `StringBuilder` here instead of all the string concatenation.

Reply:
Sure.

Reply:
BTW IntelliJ is suggesting that this should be replaced by String concatenation. But if you prefer StringBuilder no problem.